src/evo/dmnstate.cpp

```diff
@@ -39,6 +39,7 @@ void CDeterministicMNState::ToJson(UniValue& obj) const
     obj.pushKV("service", addr.ToStringIPPort(false));
     obj.pushKV("registeredHeight", nRegisteredHeight);
     obj.pushKV("lastPaidHeight", nLastPaidHeight);
+    obj.pushKV("consecutivePayments", nConsecutivePayments);
```

Comment:
haven't seen this get decreased on block rewind anywhere yet

Reply:
@PastaPastaPasta Yup good catch my friend! 
It seems that @UdjinM6 missed that :)

Reply:
> @PastaPastaPasta Yup good catch my friend! It seems that @UdjinM6 missed that :)

Doesn't sound right :thinking: I bet it was just a test to see if @PastaPastaPasta was paying attention. :smile:

Reply:
`nConsecutivePayments` is updated in `BuildNewListFromBlock` and need no rollback logic I think

Reply:
@PastaPastaPasta when we undo a block we just use another (previous) mnlist, there is nothing to undo in mnlist and/or mn states

Reply:
Gotcha, I wasn't sure if we needed any special logic, but I wanted to make sure we thought about it 👍